### PR TITLE
Feat/menu select

### PR DESCRIPTION
**Pull request by iam4x**
**Summary**

Displays a menu widget as a <select> DOM Element

**Result**

menu select

### CODE REVIEW

File: packages/react-instantsearch/src/widgets/MenuSelect.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * The MenuSelect component displays a select that lets the user choose a single value for a specific attribute.
+ * @name MenuSelect
+ * @kind widget
+ * @requirements The attribute passed to the `attributeName` prop must be present in "attributes for faceting"
+ * on the Algolia dashboard or configured as `attributesForFaceting` via a set settings call to the Algolia API.
+ * @propType {string} attributeName - the name of the attribute in the record
+ * @propType {string} [defaultRefinement] - the value of the item selected by default
+ * @propType {function} [transformItems] - Function to modify the items being displayed, e.g. for filtering or sorting them. Takes an items as parameter and expects it back in return.
+ * @themeKey ais-MenuSelect__select - the select container of all items in the
```
Comment (mthuret) [inline packages/react-instantsearch/src/widgets/MenuSelect.js]: missing the end?

File: packages/react-instantsearch/src/components/MenuSelect.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        onChange={this.handleSelectChange}
+        {...cx('select')}
+      >
+        <option value="ais__see__all__option" {...cx('option')}>
+          {translate('seeAllOption')}
+        </option>
+
+        {items.map(item => (
+          <option key={item.value} value={item.value} {...cx('option')}>
+            {item.label} ({item.count})
```
Comment (mthuret) [inline packages/react-instantsearch/src/components/MenuSelect.js]: I'm wondering if there were a solution to allow our user removing count if they want to?

File: packages/react-instantsearch/src/components/MenuSelect.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      })
+    ),
+  };
+
+  static contextTypes = {
+    canRefine: PropTypes.func,
+  };
+
+  get selectedValue() {
+    const { value } = find(this.props.items, { isRefined: true }) || {
```
[the remark this answers is not in the document]
Comment (iam4x) [reply packages/react-instantsearch/src/components/MenuSelect.js]: I like the syntax and it's already used into `connectMultiRanges` so it does not increase the size of the build.

File: packages/react-instantsearch/src/components/MenuSelect.js (hunk 2 above)
Comment (iam4x) [reply packages/react-instantsearch/src/components/MenuSelect.js]: Good idea, we can add a prop for it 👍

**Comment by iam4x** [comment]
Should be good now, but I'm not sure about the naming of the prop to customize the label output: `formatItemLabel` what do you think @mthuret?

**Comment by mthuret** [comment]
@iam4x is the feature allowed by `formatItemLabel` available on other menuSelect implementation? 

I'm wondering if this props will not be confused with `transformItems` (which is used to modify items data for instance).

What do you think @vvo @iam4x? To remove the count is this worth a prop or should we default on 'you should use a connector'.

**Comment by iam4x** [comment]
This feature is not available in other `menuSelect` implementation, to keep consistency we should remove it or add it everywhere else.

I think it will confuse, I wanted to call it `transformItems` first.

**Comment by mthuret** [comment]
Ok, so let's remove this option @iam4x. Sorry you developed it :(

**Comment by iam4x** [comment]
@mthuret Option removed, rebased on master should be good 👍

**Pull request merged**